Source/WebCore/dom/Node.cpp

```diff
@@ -2102,6 +2106,23 @@ Element* Node::enclosingLinkEventParentOrSelf()
     return nullptr;
 }
 
+void Node::expandAnyDetailsAncestor()
+{
+    if (!document().settings().detailsAutoExpandEnabled())
+        return;
+    ShouldNotFireMutationEventsScope scope(document());
+    for (auto& element : ancestorsOfType<HTMLElement>(*this)) {
+        if (is<HTMLSummaryElement>(element))
+            continue;
+        if (RefPtr detailsElement = dynamicDowncast<HTMLDetailsElement>(element)) {
+            if (!detailsElement->hasAttributeWithoutSynchronization(HTMLNames::openAttr)) {
+                detailsElement->toggleOpen();
+                break;
```

Comment:
Why does this break here? The specification seems to suggest it should expand all ancestors, not just the first with a non-matching `open` attribute.